### PR TITLE
docs: Move wiki to docusaurus for osp-backend

### LINKED ISSUE #75: doc: Move all the documentation of open-source-programs-backend to docusaurus
### Is your feature request related to a problem? Please describe.
The current documentation of open-source-programs-backend needs to be moved to docusaurus.

### Describe the solution you'd like
[Docusaurus](https://v2.docusaurus.io/) which will help get more focus on the content.

### Describe alternatives you've considered
I've considered no alternative solutions or features.

## Definition of Done
- [ ] All of the required items are completed.
- [x] Approval by 1 mentor.

### PR DESCRIPTION
**Pull request by ashwinijha6**
### Description

Moved documentation of open source backend wiki to docusaurus

Fixes #75

### Type of Change:

- Code
- Quality Assurance
- Documentation

**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
The osp backend documentation has been hosted on [https://osp-backend-docs.surge.sh](https://osp-backend-docs.surge.sh).

### Checklist:

- [ ] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] I have written Kotlin Docs whenever is applicable

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by ashwinijha6** [comment]
Could you pls review my pull request @isabelcosta 
I don't understand why my test are failing could you help me with it.

**Comment by jeeva2812** [comment]
@ashwinijha6  I think failing the test isn't a problem. I recently got a PR merged, even that was failing the test.

File: README.md
Comment (isabelcosta) [inline README.md]: there is already a section for documentation a few lines below on README, could you add this sentence there, please?

**Comment by isabelcosta** [comment]
> @ashwinijha6 I think failing the test isn't a problem. I recently got a PR merged, even that was failing the test.

yap, in this case, we have a problem with the way our tests are setup and they depend on an environment variable, apparently, these PRs do not have access to them. so the tests are not failing because of your PR @ashwinijha6  :) 
I left a note on your PR, once that is done I am happy to approve

**Review by Aaishpra — COMMENTED** [review]
Everything looks correct to me 🚀 , I can approve this when you add the above suggestion by @isabelcosta ,
And don't worry about the test which is failing, it is expected to fail (considering we cannot commit our zulip API key file )

**Comment by ashwinijha6** [comment]
> Everything looks correct to me 🚀 , I can approve this when you add the above suggestion by @isabelcosta ,
> And don't worry about the test which is failing, it is expected to fail (considering we cannot commit our zulip API key file )

Sure @Aaishpra

File: osp-backend-docusaurus/docs/Code- Organization.md
Comment (codesankalp) [inline osp-backend-docusaurus/docs/Code- Organization.md]: There are two files present with name `Code- Organization.md` and `Code-Organization.md`. Please remove the first one as that is not needed.

File: osp-backend-docusaurus/docs/Code-Organization.md
Comment (codesankalp) [inline osp-backend-docusaurus/docs/Code-Organization.md:2]: This is another `Code-Organization.md` file.

**Review by codesankalp — CHANGES_REQUESTED** [review]
Everything looks fine and will approve this @ashwinijha6  after suggested changes by me and @isabelcosta

File: README.md
Comment (ashwinijha6) [reply README.md]: hey @isabelcosta should I remove the line stating about the wiki of the backend & frontend (For more information, you can read backend project wiki and the web project wiki) & add this line or add it separately?

File: osp-backend-docusaurus/docs/Code- Organization.md
Comment (ashwinijha6) [reply osp-backend-docusaurus/docs/Code- Organization.md]: Yes I just noticed that I was just renaming the file & I guess I added it twice I will remove it

**Comment by ashwinijha6** [comment]
Could somebody pls help me this conflict in README.md?

**Comment by codesankalp** [comment]
@ashwinijha6 Use these commands in your terminal:
```
git remote add upstream git@github.com:anitab-org/open-source-programs-backend.git
git fetch upstream
git checkout docusaurus
git rebase upstream/develop
git push -f origin docusaurus
```

**Comment by Aaishpra** [comment]
@ashwinijha6 can you tell us in which lines you are having conflicts in `README.md`.

**Comment by ashwinijha6** [comment]
> @ashwinijha6 can you tell us in which lines you are having conflicts in `README.md`.

Yes @Aaishpra you can check resolve conflicts

**Comment by Aaishpra** [comment]
> Yes @Aaishpra you can check resolve conflicts

@ashwinijha6 I don't think I am allowed to see or resolve merge conflicts on your PR
here is what I see on my screen
image
I guess only the PR author is allowed to do so

**Comment by ashwinijha6** [comment]
readme

**Comment by Aaishpra** [comment]
@ashwinijha6 can you just remove/delete the `<<<<<<devlop` and `<<<<<docusaurus` line directly from Github GUI and see if it works.
If you still face any doubts about doing that check out this [video](https://www.youtube.com/watch?v=JtIX3HJKwfo)

**Comment by ashwinijha6** [comment]
> @ashwinijha6 can you just remove/delete the `<<<<<<devlop` and `<<<<<docusaurus` line directly from Github GUI and see if it works.
> If you still face any doubts about doing that check out this [video](https://www.youtube.com/watch?v=JtIX3HJKwfo)

Thanx @Aaishpra & @codesankalp  I guess I was able to resolve it just needed to delete the lines

**Comment by ashwinijha6** [comment]
> @ashwinijha6 can you just remove/delete the `<<<<<<devlop` and `<<<<<docusaurus` line directly from Github GUI and see if it works.
> If you still face any doubts about doing that check out this [video](https://www.youtube.com/watch?v=JtIX3HJKwfo)
Yes @Aaishpra  along with that I needed to delete the second line **Read the....**  just before <<<<<<<develop 😄

**Review by Aaishpra — DISMISSED** [review]
Thanks, @ashwinijha6 for the contribution :tada:
Approving this because all the suggested changes are completed

**Review by codesankalp — DISMISSED** [review]
LGTM @ashwinijha6 :tada:

File: osp-backend-docusaurus/docs/Getting-Started.md
Comment (Rahulm2310) [inline osp-backend-docusaurus/docs/Getting-Started.md]: @ashwinijha6 Can you remove this link to wiki and add the link to the `Commit Message Style Guide.md` file present locally in `osp-backend-docusaurus/docs`.

File: osp-backend-docusaurus/docs/Getting-Started.md
Comment (Rahulm2310) [inline osp-backend-docusaurus/docs/Getting-Started.md]: You can add links to `[Fork](./Fork,-Clone,-Remote-and-Pull-Request.md#Fork)`, similarly to `Clone` and `Setup remotes`.

File: osp-backend-docusaurus/docs/Maintainer-Guidelines.md
Comment (Rahulm2310) [inline osp-backend-docusaurus/docs/Maintainer-Guidelines.md]: Please update the link to `Commit Message Style Guide` here as well as mentioned above.

File: osp-backend-docusaurus/docusaurus.config.js
Comment (Rahulm2310) [inline osp-backend-docusaurus/docusaurus.config.js]: It should be `AnitaB.org` and remove `Inc`.

File: osp-backend-docusaurus/docusaurus.config.js
Comment (Rahulm2310) [inline osp-backend-docusaurus/docusaurus.config.js]: Please remove this `blog` object as it is not needed.

Comment (Rahulm2310) [inline osp-backend-docusaurus/docusaurus.config.js]: Please remove this `blog` object as it is not needed.

File: osp-backend-docusaurus/docusaurus.config.js
Comment (Rahulm2310) [inline osp-backend-docusaurus/docusaurus.config.js]: Please correct this `editUrl` to `https://github.com/anitab-org/open-source-programs-backend/osp-backend-docusaurus`.

File: osp-backend-docusaurus/src/pages/index.js
Comment (Rahulm2310) [inline osp-backend-docusaurus/src/pages/index.js]: You can add a description for the project here.

File: osp-backend-docusaurus/src/pages/styles.module.css
Comment (Rahulm2310) [inline osp-backend-docusaurus/src/pages/styles.module.css]: Please remove the `.feature` and `.featureImage` styles as they are not needed.

**Comment by ashwinijha6** [comment]
@isabelcosta  & @Rahulm2310  I have made the required changes, pls review my pull request

File: osp-backend-docusaurus/docs/Getting-Started.md
Comment (Rahulm2310) [inline osp-backend-docusaurus/docs/Getting-Started.md]: You can add relative path to `commit message style guide` file here.

File: osp-backend-docusaurus/docs/Maintainer-Guidelines.md
Comment (Rahulm2310) [inline osp-backend-docusaurus/docs/Maintainer-Guidelines.md]: add a relative path here as well

File: osp-backend-docusaurus/docusaurus.config.js
Comment (Rahulm2310) [inline osp-backend-docusaurus/docusaurus.config.js]: It should be `AnitaB.org`.

**Comment by Rahulm2310** [comment]
Also, as now we get page titles with docusaurus, the existing headings looks redundant. Shouldn't we remove them.  
image
image

What say @codesankalp @isabelcosta ?

**Comment by ashwinijha6** [comment]
Yes @Rahulm2310  we could remove it as it looks redunant

**Comment by codesankalp** [comment]
I will approve this PR after the suggested changes will be fixed @ashwinijha6.

**Comment by ashwinijha6** [comment]
I'm sorry for the delay ran into a `grub` issue 
Pls review the changes @isabelcosta , @codesankalp  &  @Rahulm2310

**Comment by codesankalp** [comment]
@Rahulm2310 Can you review it again?

**Review by jeeva2812 — APPROVED** [review]
LGTM. Great Work. :rocket:

**Review by Rahulm2310 — APPROVED** [review]
LGTM 💯 . Thanks @ashwinijha6 for applying suggested changes. 
@codesankalp I think it is ready to merge.

**Review by codesankalp — APPROVED** [review]
Thanks for your contribution @ashwinijha6 👍 🎉 .

**Review by Aaishpra — APPROVED** [review]

**Comment by codesankalp** [comment]
@isabelcosta Tagging you for final review.

**Review by isabelcosta — APPROVED** [review]
Thank you so much for working on this @ashwinijha6 :) 🎉

**Comment by ashwinijha6** [comment]
> Thank you so much for working on this @ashwinijha6 :) 🎉

Thankyou everyone for helping me out with this 😄

**Pull request merged**
